tsfresh/feature_extraction/extraction.py

```diff
@@ -76,23 +85,25 @@ def extract_features(timeseries_container, feature_extraction_settings=None,
         for key in kind_to_df_map:
             feature_extraction_settings.set_default_parameters(key)
 
+    # Choose the parallelization according to a rule-of-thumb
+    if parallelization is None and (feature_extraction_settings.n_processes / 2) > len(kind_to_df_map):
```

Comment:
The idea behind the rule is exactly this: If I have more cores it might be worth to parallelize by sample. But only if the cores wouldn't be busy anyway by parallelizing by kind.
A better rule should also incorporate the number of samples and the length of the time series.

The rule as it is now maps with my experiments while developing this branch. However, we could also just use parallelization per kind as default and leave it up to the user to enable paralleliziation per sample.

Logging the resulting parallelization is a good idea. I'll add it.